src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,356 @@
+        return tasks.size();
+    }
+
+    //retrieve past saved data
+    private static void handlePastData() {
+        File pastFile = new File(filepath); // create a File for the given file path
+        String wholeDoc = readFile(pastFile);
+        //when file is empty or not created
+        if (wholeDoc == null) {
+            return;//skip parsing data
+        }
+        //convert file entry to tasks
+        parseFile(wholeDoc);
+    }
+
+    //read file content
+    private static String readFile(File pastFile) {
+        String wholeDoc = null;
+        try {
+            //read previous data
+            Scanner s = new Scanner(pastFile);
+            while (s.hasNext()) {
+                wholeDoc = s.nextLine();
+            }
+        } catch (FileNotFoundException e) {
+            printFileNotFound();
+            return null;
+        }
+        return wholeDoc;
+    }
+
+    //display file not found error message
+    private static void printFileNotFound() {
+        System.out.println(" Are you first time user?\n" +
+                " I am unable to find your past data\n" +
+                "____________________________________________________________");
+    }
+
+    //convert file entry to tasks
+    private static void parseFile(String wholeDoc) {
+        //separate entry based on ||
+        String[] taskListFile = wholeDoc.split("\\|\\|", MAX_LIST_SIZE);
+        for (int i = 0; i < taskListFile.length - 1; i++) {//minus one to negate for last || on file
+            String[] taskDetails = taskListFile[i].split("\\|", MAX_TASK_LENGTH);
+            switch (taskDetails[0]) {
+            case "T":
+                addToDoFile(taskDetails);
+                break;
+            case "E":
+                addEventFile(taskDetails);
+                break;
+            case "D":
+                addDeadlineFile(taskDetails);
+                break;
+            default:
+                System.out.println("Error occur on retrieving previous data");
+            }
+        }
+    }
+
+    //add deadline from file
+    private static void addDeadlineFile(String[] taskDetails) {
+        Deadline deadline = new Deadline(taskDetails[2], TaskType.DEADLINE, taskDetails[3]);
+        tasks.add(deadline);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+    //add event from file
+    private static void addEventFile(String[] taskDetails) {
+        Event event = new Event(taskDetails[2], TaskType.EVENT, taskDetails[3]);
+        tasks.add(event);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+    //add todo from file
+    private static void addToDoFile(String[] taskDetails) {
+        ToDo toDo = new ToDo(taskDetails[2], TaskType.TODO);
+        tasks.add(toDo);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
```

Comment:
All 3 functions here are doing the same thing, can consider refactoring into one class so that you don't have to repeat what you wrote 3x